server/admin/admin_server_test.go

```diff
@@ -562,7 +562,9 @@ func TestServer_UpdateTree(t *testing.T) {
 		StorageSettings: settings,
 		MaxRootDuration: ptypes.DurationProto(2 * time.Nanosecond),
 	}
-	successMask := &field_mask.FieldMask{Paths: []string{"tree_state", "display_name", "description", "storage_settings", "max_root_duration"}}
+	successMask := &field_mask.FieldMask{
+		Paths: []string{"tree_state", "display_name", "description", "storage_settings", "max_root_duration", "private_key"},
```

Comment:
I think you need a corresponding change in successTree, otherwise I'd expect it to error.

Reply:
Because private_key is redacted from the response, it doesn't need a change to successTree.

Reply:
Because successTree and successMask are paired together in the update request, I expected successTree to have a non-nil PrivateKey, otherwise applying the mask would just "nil" the old key.

The test is heavily mocked, so I think that's why it works. You might get better value by adding this to the admin integration test instead.

Reply:
Ah yes, and I see that `successWant.PrivateKey = nil // redacted on responses` is also misleading, since in the test `applyUpdateMask()` is run **after** `UpdateTree()` has returned, which would override the redaction done by `UpdateTree()`. I've fixed the test so that the tree is updated at the appropriate time and the redaction still takes place.